tests/keras/engine/test_topology.py

```diff
@@ -656,6 +656,28 @@ def test_recursion_with_bn_and_loss():
     model2.fit(x, y, verbose=0, epochs=1)
 
 
+@keras_test
+def test_activity_regularization_with_model_composition():
+    # Ensures that the shape of the input to the activity
+    # regularizer is the same under model composition.
+    # Tests for regressions of issue #9267
+    wrong_size = 10
+    correct_size = 2
+    def reg(x):
+        assert x.shape[-1] == correct_size
```

Comment:
Is this really the right way to test activity regularization?

Reply:
Not sure exactly what you mean - it tests a specific property that should be true, but it's not a very general one? Happy to go read some better tests and re-write this if you can suggest an example that'll be informative :)

Reply:
> it tests a specific property that should be true, but it's not a very general one

Basically this: you ran into a bug under a specific manifestation, and you are testing that the specific symptoms you encountered aren't there. But it's not testing that there the bug is fixed. I didn't give you a suggestion because I didn't immediately have one.

I would say this:

- build model with single Dense layer with 2 units, `initializer='ones', bias=False`, `activity_regularizer` that sums all values
- call model on `K.ones` of shape `(1, 2)` (wrapped in `Input`)
- `loss=None` in `compile`
- compute loss via `model.evaluate`, check it is equal to 4